fe/fe-core/src/main/java/org/apache/doris/catalog/Catalog.java

```diff
@@ -5045,6 +5046,36 @@ public void renameTable(Database db, OlapTable table, TableRenameClause tableRen
         LOG.info("rename table[{}] to {}", tableName, newTableName);
     }
 
+    public void renameTable(Database db, Table table, TableRenameClause tableRenameClause) throws DdlException {
+        String tableName = table.getName();
+        String newTableName = tableRenameClause.getNewTableName();
+        if (tableName.equals(newTableName)) {
+            throw new DdlException("Same table name");
+        }
+
+        // check if name is already used
+        if (db.getTable(newTableName) != null) {
+            throw new DdlException("Table name[" + newTableName + "] is already used");
+        }
+
+        table.setName(newTableName);
+
+        db.dropTable(tableName);
+        db.createTable(table);
+
+        TableInfo tableInfo = TableInfo.createForTableRename(db.getId(), table.getId(), newTableName);
+        editLog.logTableRename(tableInfo);
+        LOG.info("rename table[{}] to {}", tableName, newTableName);
+    }
+
+    public void reflushTable(Database db, Table table) throws DdlException {
```

Comment:
This is not a good way to alter external table.
You write 2 edit logs here, which will make this operation non-atomic。

Reply:
yes，to make sure it is atomic. We may need add a new OperationType in edit log.